### ISSUE TITLE
Option to toggle a sign that indicate lines that diff against X branch

### ISSUE THREAD

**Issue by zapling** [lewis6991/gitsigns.nvim#407]
**Is your feature request related to a problem? Please describe.**
When working on a feature branch and committing and then coming back a few days later, it would be nice to see which lines that diff against the main branch. Now I have to do `git diff origin/main`. 

**Describe the solution you'd like**
Able to toggle a command to add a sign in the column to indicate that the line differs from X branch.

**Describe alternatives you've considered**
No other solution than using `git diff origin/main`

**Additional context**
So basically I would like to have a configurable sign like this indicating the lines that diff against X branch.
```
>   // this comment is not on origin/main
>   // not this comment either
    // this comment is on origin/main
    // this one as well
```

**Comment by lewis6991** [comment]
Doesn't `:Gitsigns change_base` fulfill this?

**Comment by zapling** [comment]
Ah yes it does give the the info I need, you have thought of everything! Thank you!
